src/lib/proxy-server.ts

```diff
@@ -97,14 +97,16 @@ export async function startProxyServer(): Promise< boolean > {
 		return true;
 	} catch ( error ) {
 		if ( error instanceof Error && 'code' in error && error.code === 'EADDRINUSE' ) {
-			dialog.showMessageBox( {
-				type: 'error',
-				message: __( 'Failed to start custom domain proxy server' ),
-				detail: __(
-					'Another server is already running on port 80. For custom domains to work, please stop that server and then restart Studio.'
-				),
-				buttons: [ __( 'OK' ) ],
-			} );
+			setTimeout( () => {
+				dialog.showMessageBox( {
+					type: 'error',
+					message: __( 'Custom domain set up failed' ),
+					detail: __(
+						'Studio needs to use port 80, but it’s in use by another app. Close any local development apps and restart Studio.'
+					),
+					buttons: [ __( 'OK' ) ],
+				} );
+			}, 2000 );
```

Comment:
```suggestion
			const mainWindow = await getMainWindow();
			dialog.showMessageBox( mainWindow, {
				type: 'error',
				message: __( 'Custom domain set up failed' ),
				detail: __(
					'Studio needs to use port 80, but it’s in use by another app. Close any local development apps and restart Studio.'
				),
				buttons: [ __( 'OK' ) ],
			} );
```
I realized that passing a window instance to `dialog.showMessageBox` unblocks the rendering. I think this approach is preferable since a timeout is inherently arbitrary.

Reply:
That's very nice. Thanks for your review and feedback, Fredrik. I have applied your suggestion in https://github.com/Automattic/studio/pull/1068/commits/da29638d61ae362b2d4bcd4513c77ac5df61f120.